### PR TITLE
Roland/deploy docker

### PR DESCRIPTION
**Pull request by RolandNaijuka**
- Run docker from an image already built with node 12 and java 11
- Install dependency openapi which depends on java
- remove the duplicates from package.json
- define port on 8080

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
 COPY biogrid/package*.json ./
 
 # Install all required packages for build
 RUN npm install --ignore-scripts
 
+# Install open api for combined both back and front end
+RUN npm install @openapitools/openapi-generator-cli -g
+
+# Install open api for combined both back and front end
+# RUN npm install @openapitools/openapi-generator-cli -g
```
Comment (cwlarson-google) [inline Dockerfile]: delete?

File: Dockerfile [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
+
 # Copy in the necessary files for tsoa
 COPY biogrid/apps apps
 COPY biogrid/tsconfig.json .
 
-# Build tsoa 
-RUN npm run build:tsoa
+# ENV PORT=8080
```
Comment (cwlarson-google) [inline Dockerfile]: delete?

File: biogrid/apps/biogrid-mvp/src/main.tsx [hunk 3]
```diff
@@ -3,4 +3,11 @@ import ReactDOM from 'react-dom';
 
 import App from './app/app';
 
+declare global {
+  interface GlobalFetch {
+    [k: string]: any;
+  }
+}
+window.globalThis = window
```
Comment (cwlarson-google) [inline biogrid/apps/biogrid-mvp/src/main.tsx:11]: what's the point of this?

**Review by cwlarson-google — APPROVED** [review]

File: biogrid/apps/biogrid-mvp/src/main.tsx (hunk 3 above)
Comment (RolandNaijuka) [reply biogrid/apps/biogrid-mvp/src/main.tsx:11]: > what's the point of this?

This solves an error at runtime caused by one of the built files by tsoa.

**Pull request merged**